lib/signs/utilities/messages.ex

```diff
@@ -258,6 +289,41 @@ defmodule Signs.Utilities.Messages do
     end
   end
 
+  defp overnight_period?(
+         current_time,
+         first_scheduled_departure,
+         last_scheduled_departure,
+         recent_departures,
+         service_ended
+       ) do
+    last_actual_departure =
+      if service_ended do
+        recent_departures
+        |> Map.values()
```

Comment:
I think Map.values() returns the values based on the order of the keys (`trip_ids` in this case) which I'm not confident would guarantee ordering for us, considering we often get added trips with ids like `ADDED-12345`. I think we may need to sort the values before acting on the ordering of the departures